content/IA-update_connections/connection-state-and-recovery.textile

```diff
@@ -0,0 +1,292 @@
+
+- closing := An explicit request by the developer to close the connection has been sent to the Ably service. If a reply is not received from Ably within a short period of time, the connection will be forcibly terminated and the connection state will become @closed@.
+
+- closed := The connection has been explicitly closed by the client.<br><br>In the closed state, no reconnection attempts are made automatically by the library, and clients may not publish messages. No connection state is preserved by the service or by the library. A new connection attempt can be triggered by an explicit call to "<span lang="default">@connect()@</span><span lang="ruby">@connect@</span><span lang="csharp">@Connect()@:#connect</span>":/api/realtime-sdk/connection#connect on the @Connection@ object, which will result in a new connection.
+
+- failed := This state is entered if the client library encounters a failure condition that it cannot recover from. This may be a fatal connection error received from the Ably service (e.g. an attempt to connect with an incorrect API key), or some local terminal error (e.g. the token in use has expired and the library does not have any way to renew it).<br><br>In the failed state, no reconnection attempts are made automatically by the library, and clients may not publish messages. A new connection attempt can be triggered by an explicit call to "<span lang="default">@connect()@</span><span lang="ruby">@connect@</span><span lang="csharp">@Connect()@:#connect</span>":/api/realtime-sdk/connection#connect on the @Connection@ object.
+
+h4. Typical connection state sequences
+
+The library is initialized and initiates a successful connection.
+
+bq. initialized &rarr; connecting &rarr; connected
+
+An existing connection is dropped and reestablished on the first attempt.
+
+bq. connected &rarr; disconnected &rarr; connecting &rarr; connected
+
+An existing connection is dropped, and reestablished after several attempts but within a two minute interval.
+
+bq. connected &rarr; disconnected &rarr; connecting &rarr; disconnected &rarr; &hellip; &rarr; connecting &rarr; connected
+
+There is no connection established after initializing the library.
+
+bq. initialized &rarr; connecting &rarr; disconnected &rarr; connecting &rarr; &hellip; &rarr; suspended
+
+After a period of being offline a connection is reestablished.
+
+bq. suspended &rarr; connecting &rarr; suspended &rarr; &hellip; &rarr; connecting &rarr; connected
+
+h4. Listening for state changes
```

Comment:
This should be "Listen for connection state change". This should also be an h2.

Reply:
Fixed